### PR TITLE
fix(633): persist run image_url via repository, not raw SQL (#715)

### PR DESCRIPTION
**Pull request by axisrow**
## Что и зачем

Закрывает суб-issue #715 (баг #26 из мета-issue #633): `ContentGenerationService.generate` дважды писал raw SQL `UPDATE generation_runs SET image_url = ? WHERE id = ?` через `execute_write`, минуя слой репозиториев и не обновляя `updated_at` (в отличие от всех остальных сеттеров `generation_runs`).

## Решение

- Новый метод `GenerationRunsRepository.set_image_url(run_id, image_url)` — обновляет `image_url` и `updated_at = datetime('now')`, как остальные сеттеры.
- Оба call-site в сервисе переведены на `db.repos.generation_runs.set_image_url(...)`.

## Тесты

- `tests/repositories/test_generation_runs_repository.py::test_set_image_url_updates_row_and_timestamp`.
- Обновлён `test_generate_image_url_from_graph_executor` (проверка через `repo.get(...).image_url`); `FakeGenerationRunsRepo` получил `set_image_url`.

## Проверка

- `ruff check` — чисто
- 453 связанных теста зелёные (integration_generation / unified_dispatcher / content gen edge cases / repositories). Дифф 27 строк.

Refs #633

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: tests/repositories/test_generation_runs_repository.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+@pytest.mark.anyio
+async def test_set_image_url_updates_row_and_timestamp(db):
+    repo = db.repos.generation_runs
+    run_id = await repo.create_run(42, "prompt-template")
+
+    before = await repo.get(run_id)
+    assert before is not None
+    assert before.image_url is None
+
```
[the remark this answers is not in the document]
Comment (axisrow) [reply tests/repositories/test_generation_runs_repository.py:17]: Fixed in `e60bd6b`: the repository test now explicitly asserts `before.updated_at is None` before calling `set_image_url()`, so the timestamp bump precondition cannot silently drift.

**Comment by axisrow** [comment]
@codex @claude review

Focus on critical issues: bugs, security vulnerabilities, logical errors, data loss risks, performance problems. Do NOT nitpick style, naming conventions, minor formatting, or subjective preferences — only flag issues that could break functionality or cause real harm in production.

Context: addressed Copilot's inline test precondition comment in `e60bd6b`; target repository/service tests and Ruff are passing locally.
